apps/console/src/features/connections/api/connections.ts

```diff
@@ -0,0 +1,1205 @@
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Failed to update identity provider: " + idpId));
+            }
+
+            return Promise.resolve(response.data as ConnectionInterface);
+        }).catch((error: AxiosError) => {
+            return Promise.reject(error);
+        });
+};
+
+/**
+ * Update JIT provisioning configs of a specified IDP.
+ *
+ * @param idpId - ID of the Identity Provider.
+ * @param configs - JIT provisioning configs.
+ * @returns A promise containing the response.
+ */
+export const updateJITProvisioningConfigs = (
+    idpId: string,
+    configs: JITProvisioningResponseInterface
+): Promise<ConnectionInterface> => {
+
+    const requestConfig: RequestConfigInterface = {
+        data: configs,
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.PUT,
+        url: store.getState().config.endpoints.identityProviders + "/" + idpId +
+            "/provisioning/jit"
+    };
+
+    return httpClient(requestConfig)
+        .then((response: AxiosResponse) => {
+            if (response.status !== 200) {
+                return Promise.reject(new Error("Failed to update jit configuration: " + idpId));
+            }
+
+            return Promise.resolve(response.data as ConnectionInterface);
+        }).catch((error: AxiosError) => {
+            throw new IdentityAppsApiException(
+                ConnectionManagementConstants.CONNECTION_JIT_PROVISIONING_UPDATE_ERROR,
+                error.stack,
+                error.code,
+                error.request,
+                error.response,
+                error.config
+            );
+        });
+};
+
+/**
+ * Get connected apps of the IDP.
+ *
+ * @param idpId - ID of the Identity Provider.
+ * @returns  A promise containing the response.
+ */
+export const getConnectedApps = (idpId: string): Promise<any> => {
+
+    const requestConfig: RequestConfigInterface = {
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        url: store.getState().config.endpoints.identityProviders + "/" + idpId + "/connected-apps/"
+    };
+
+    return httpClient(requestConfig)
+        .then((response: AxiosResponse) => {
+            if (response.status !== 200) {
+                return Promise.reject(
+                    new Error("Failed to get connected apps for the IDP: " + idpId)
+                );
+            }
+
+            return Promise.resolve(response.data as ConnectedAppsInterface);
+        }).catch((error: AxiosError) => {
+            return Promise.reject(error);
+        });
+};
+
+/**
+ * Update identity provider details.
+ *
+ * @param connection - Connection.
+ * @returns A promise containing the response.
+ */
+export const updateIdentityProviderDetails = (connection: ConnectionInterface): Promise<any> => {
```

Comment:
Shouldn't the method name be changed to use the term "Connection"?

Reply:
As these changes are directly brought from previous PRs, will continue as it is.